tests/snap_test.go

```diff
@@ -215,22 +227,35 @@ func TestWifiMatterCommander(t *testing.T) {
 	var stdout, stderr string
 	var err error
 
+	start := time.Now()
+
+	t.Cleanup(func() {
+		utils.SnapStop(t, snapMatterPiGPIO)
+
+		if env.Teardown() {
+			// Remove snaps, ignore errors during removal
+			utils.SnapRemove(nil, chipToolSnap)
+			// snapMatterPiGPIO is not removed here, as that is handled by the teardown function
+		}
```

Comment:
I agree, but this is the logic that was originally used in these tests. I assume it is because gpio-commander is modified for mocked gpio, and we only want to do that, and the installation once.

Reply:
I re-read your comment. I think I understand now. I'll break this one cleanup into individual ones.